Sources/NIOCore/IPProtocol.swift

```diff
@@ -0,0 +1,119 @@
+    public static let dccp = Self(rawValue: 33)
+    /// IPv6 encapsulation - [RFC2473]
+    public static let ipv6 = Self(rawValue: 41)
+    /// Reservation Protocol - [RFC2205][RFC3209][Bob_Braden]
+    public static let rsvp = Self(rawValue: 46)
+    /// Generic Routing Encapsulation - [RFC2784][Tony_Li]
+    public static let gre = Self(rawValue: 47)
+    /// Dynamic Source Routing Protocol - [RFC4728]
+    public static let dsr = Self(rawValue: 48)
+    /// Encap Security Payload - [RFC4303]
+    public static let esp = Self(rawValue: 50)
+    /// Authentication Header - [RFC4302]
+    public static let ah = Self(rawValue: 51)
+    /// NBMA Address Resolution Protocol - [RFC1735]
+    public static let narp = Self(rawValue: 54)
+    /// ICMP for IPv6 - [RFC8200]
+    public static let ipv6Icmp = Self(rawValue: 58)
+    /// No Next Header for IPv6 - [RFC8200]
+    public static let ipv6Nonxt = Self(rawValue: 59)
+    /// Destination Options for IPv6 - [RFC8200]
+    public static let ipv6Opts = Self(rawValue: 60)
+    /// EIGRP - [RFC7868]
+    public static let eigrp = Self(rawValue: 88)
+    /// OSPFIGP - [RFC1583][RFC2328][RFC5340][John_Moy]
+    public static let ospfigp = Self(rawValue: 89)
+    /// Ethernet-within-IP Encapsulation - [RFC3378]
+    public static let etherip = Self(rawValue: 97)
+    /// Encapsulation Header - [RFC1241][Robert_Woodburn]
+    public static let encap = Self(rawValue: 98)
+    /// Protocol Independent Multicast - [RFC7761][Dino_Farinacci]
+    public static let pim = Self(rawValue: 103)
+    /// IP Payload Compression Protocol - [RFC2393]
+    public static let ipcomp = Self(rawValue: 108)
+    /// Virtual Router Redundancy Protocol - [RFC5798]
+    public static let vrrp = Self(rawValue: 112)
+    /// Layer Two Tunneling Protocol - [RFC3931][Bernard_Aboba]
+    public static let l2tp = Self(rawValue: 115)
+    /// Fibre Channel - [Murali_Rajagopal][RFC6172]
+    public static let fc = Self(rawValue: 133)
+    /// MANET Protocols - [RFC5498]
+    public static let manet = Self(rawValue: 138)
+    /// Host Identity Protocol - [RFC7401]
+    public static let hip = Self(rawValue: 139)
+    /// Shim6 Protocol - [RFC5533]
+    public static let shim6 = Self(rawValue: 140)
+    /// Wrapped Encapsulating Security Payload - [RFC5840]
+    public static let wesp = Self(rawValue: 141)
+    /// Robust Header Compression - [RFC5858]
+    public static let rohc = Self(rawValue: 142)
+    /// Ethernet - [RFC8986]
+    public static let ethernet = Self(rawValue: 143)
+    /// AGGFRAG encapsulation payload for ESP - [RFC-ietf-ipsecme-iptfs-19]
+    public static let aggfrag = Self(rawValue: 144)
+    // Use for experimentation and testing (253) - [RFC3692]
+    public static let useForExperimentationAndTesting1 = Self(rawValue: 253)
+    // Use for experimentation and testing (254) - [RFC3692]
+    public static let useForExperimentationAndTesting2 = Self(rawValue: 254)
```

Comment:
I'd skip these bottom two names: we can just leave them undefined.

Reply:
Done in 904c5fd6380eb9e5667378a39176cab829e2361f